src/impy/util.py

```diff
@@ -214,6 +221,88 @@ def readOutput(self):
             self.capturedtext += char
 
 
+# Functions to check and download dababase files on github
+
+
+def _download_file(outfile, url):
+    """Download a file from 'url' to 'outfile'"""
+    fname = pathlib.Path(url).name
+    try:
+        response = urllib.request.urlopen(url)
+    except BaseException:
+        raise ConnectionError(
+            f"_download_file: probably something wrong with url = '{url}'"
+        )
+    total_size = response.getheader("content-length")
+
+    min_blocksize = 4096
+    if total_size:
+        total_size = int(total_size)
+        blocksize = max(min_blocksize, total_size // 100)
+    else:
+        blocksize = min_blocksize
+
+    wrote = 0
+    with open(outfile, "wb") as f:
+        while True:
+            data = response.read(blocksize)
+            if not data:
+                break
+            f.write(data)
+            wrote += len(data)
+            if total_size:
+                print(
+                    f"Downloading {fname}: {wrote/total_size*100:.0f}% "
+                    f"done ({wrote/(1024*1024):.0f} Mb) \r",
+                    end=" ",
+                )
+            else:
+                print(
+                    f"Downloading {fname}: {wrote/(1024*1024):.0f} Mb downloaded \r",
+                    end=" ",
+                )
+    print()
+    if total_size and wrote != total_size:
+        raise ConnectionError(f"{fname} has not been downloaded")
+    return True
+
+
+def _cached_data_dir(url):
+    """Checks for existence of version file
+    "model_name_vxxx.zip". Downloads and unpacks
+    zip file from url in case the file is not found
+
+    Args:
+        url (str): url for zip file
+    """
+    impy_path = pathlib.Path(__file__).parent.absolute()
+    base_dir = pathlib.Path(impy_path) / "iamdata"
+    base_dir.mkdir(parents=True, exist_ok=True)
+
+    vname = Path(url).stem
+    model_dir = base_dir / vname.split("_v")[0]
+    version_file = model_dir / vname
```

Comment:
Is this really necessary?

Instead of `qgsjet_v001` we can put the specific version into the name e.g. `qgsjet203`. We can be sure that the models will always change their files and version number together, so this is safe. I don't see the need foraAdding an extra separate versioning from the data files, I mean the `_v001`.